core/src/main/scala/org/http4s/parser/Rfc3986Parser.scala

```diff
@@ -144,6 +144,6 @@ private[http4s] trait Rfc3986Parser
 
   def SubDelims = rule { "!" | "$" | "&" | "'" | "(" | ")" | "*" | "+" | "," | ";" | "=" }
 
-  protected def decode(s: String) = URLDecoder.decode(s, charset.name)
+  protected def decode(s: String) = UrlCodingUtils.urlDecode(s, charset)
```

Comment:
I'm afraid this might break tests in the query string, which _do_ encode `+`.  I was thinking of replacing the `decode` calls in `UserInfo.Parser`.  But if it works, :+1:, because this is correct in more cases.

Reply:
Ok I see, I think I should correct this, even if the ci passes, because I am not sure that explicit test cases for '+' exists ...

Reply:
I confirmed that the query is _not_ using that function.  It's path and regname.